hastexo/migrations/0010_add_user_foreign_key.py

```diff
@@ -1,10 +1,30 @@
 from django.conf import settings
-from django.db import migrations, models
+from django.db import migrations, models, OperationalError
 import django.db.models.deletion
 
 
 class Migration(migrations.Migration):
 
+    def check_stacks(apps, schema_editor):
+        """
+        Check if all stacks can be linked to a real user account.
+        If not, error out with a suggestion how to proceed.
+        """
+        Stack = apps.get_model("hastexo", "Stack")
+        AnonymousUserId = apps.get_model("student", "AnonymousUserId")
+        problematic_stacks = Stack.objects.exclude(
+            student_id__in=AnonymousUserId.objects.all().values(
+                'anonymous_user_id'))
```

Comment:
That's a good approach. I *think* you can rewrite this such that the lazily evaluated `QuerySet` becomes even more efficient, like so (note that this is untested):

```python
anonymous_user_ids = AnonymousUserId.objects.values('anonymous_user_id').distinct()
problematic_stacks = Stack.objects.exclude(student_id__in=anonymous_user_ids)
```

(From memory, `objects.values()` is the same as `objects.all().values()`)

Reply:
I don't think assigning it to a variable will cause the first queryset to be evaluated. The ORM will still create one big query from these two lines. 
You can change that to:
```python
anonymous_user_ids = list(AnonymousUserId.objects.values('anonymous_user_id').distinct())
problematic_stacks = Stack.objects.exclude(student_id__in=anonymous_user_ids)
```

Reply:
Ack, I'll run my local test steps with this and update the patch.

Reply:
using `list()` resulted in all (or at least a very large number of) stacks being excluded, I updated the patch without it, using @fghaas example.